src/niamoto/core/plugins/transformers/chains/reference_resolver.py

```
@@ -0,0 +1,236 @@
+                    )
+
+                field_value = value[field_name]
+
+                # Apply index
+                try:
+                    index = int(index_str)
+                    value = field_value[index]
+                except (ValueError, IndexError, TypeError) as e:
+                    raise ValueError(
+                        f"Invalid index '{index_str}' for field '{field_name}': {str(e)}"
+                    )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)